### ISSUE TITLE
Separate collector deployment without an ingester

### ISSUE THREAD

**Issue by TomL-dev** [jaegertracing/jaeger-operator#1505]
## Requirement - what kind of business use case are you trying to solve?

Doing distributed tracing useing jaeger over several (workload)clusters in kubernetes, and have a single observability cluster . In this observaliblity cluster we want to store all the workload clusters jaeger data. by leveraging kafka to disconnect the clusters

## Problem - what in Jaeger blocks you from solving the requirement?

Currently the assumption of the `jaeger-operator` deployment is that the `collector` and `ingester` are deployed at the same time in the same namespace/cluster. 

We want to have several workload clusters to run applications and use the `jaeger agent` (auto injected) and the `collector` to push tracing data to kafka. We then want a separate cluster which will run kafka and the `ingester` and the datastore or our choice. But we currently cant deploy the `collector` without providing `ingester` configuration (which will also deploy a `ingester`)

## Proposal - what do you suggest to solve the problem or improve the existing situation?

seperate the `ingester` and `collector` deployments, so you can deploy one without the other

**Comment by jpkrohling** [comment]
Do you think adding flags like `skip-query`, `skip-collector` and `skip-ingester` would be sufficient for your use case?

**Comment by TomL-dev** [comment]
i do still feel like the operator needs ingester config to deploy a collector, but this should be checked ofc.
if a skip flag can prevent a deploy then it should be fine, if the config is separate

**Comment by jpkrohling** [comment]
Not quite sure I get your point, could you please expand it? From what I understood, you'd need one CR per workload cluster, each with a "skip-ingester". Your collectors would write to a Kafka somewhere. On your observability cluster, you'd have a single CR, with "skip-collector", which would read from Kafka and write to the storage. Your Jaeger UI connects to the same storage that the ingester writes to.

**Comment by TomL-dev** [comment]
exactly. but currently the collector also depends on ingester configuration, at least that seems like it. i don't want to provide ingester config if i don't need an ingester, right? there is also the enabled option in the config. when ingester if disabled, the collector doesn't seem to work/deploy. i hope this explains it

**Comment by jpkrohling** [comment]
Got it: you are talking about the current behavior, and I'm talking about the proposed new behavior. I think we are aligned here :-)
